tests/testenvironment.proj

```diff
@@ -79,16 +101,6 @@
     <TestEnvironment Include="jitstress2_jitstressregs0x80" JitStress="2" JitStressRegs="0x80" />
     <TestEnvironment Include="jitstress2_jitstressregs0x1000" JitStress="2" JitStressRegs="0x1000" />
     <TestEnvironment Include="tailcallstress" TailcallStress="1" />
-    <TestEnvironment Include="jitsse2only" EnableAVX="0" EnableSSE3_4="0" />
```

Comment:
These removed jobs are now covered by the above, which is more expansive. Where there are fewer flags specified, it is because the `Enable{ISA}=0` flags are hierarchical in nature.
* These "scenario names" only looked to be referenced from the `netci.groovy` and the `ci-trigger-phrases.md` documentation (the latter of which should probably be updated for AzDO).

For example, `jitstressisas_nosse3_4` is the exact replacement for `jitsse2only` since `EnableSSE3_4=0` also implicitly sets `EnableAVX=0`.

Likewise, `jitstressisas_nohwintrinsicx86` is the replacement for `jitnox86hwintrinsic` but most ISAs are implicitly disabled by `EnableSSE=0`

Reply:
We also no longer need `EnableIncompleteISAClass` for most jobs since the x86 intrinsics are "feature complete" at this point.